packages/manager/modules/pci/src/projects/project/storages/databases/database/metrics/metrics.controller.js

```diff
@@ -0,0 +1,137 @@
+import map from 'lodash/map';
+import sortBy from 'lodash/sortBy';
+import {
+  METRICS_TIME_RANGES,
+  CHART_METRICS_OPTIONS,
+  CHART_METRICS_REFRESH_INTERVAL,
+} from '../../databases.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $anchorScroll,
+    $translate,
+    $q,
+    CucCloudMessage,
+    DatabaseService,
+    PciChartjsFactory,
+    $scope,
+  ) {
+    this.$anchorScroll = $anchorScroll;
+    this.$translate = $translate;
+    this.$q = $q;
+    this.CucCloudMessage = CucCloudMessage;
+    this.DatabaseService = DatabaseService;
+    this.PciChartjsFactory = PciChartjsFactory;
+    this.isLoading = true;
+    [this.selectedTimeRange] = METRICS_TIME_RANGES;
+    this.timeRanges = METRICS_TIME_RANGES;
+    this.metricsData = {};
+    this.autoRefresh = false;
+    this.$scope = $scope;
+  }
+
+  $onInit() {
+    this.messageContainer =
+      'pci.projects.project.storages.databases.dashboard.metrics';
+    this.loadMessages();
+    this.trackDatabases('dashboard::metrics', 'page');
+
+    this.isLoading = false;
+
+    this.selectedTimeRange.label = this.$translate.instant(
```

Comment:
**Suggestion:** The chart data build up may be moved into a separate method to make it more readable?

Reply:
Done